src/main/steps/c100-rebuild/applicant/contact-preference/postController.ts

```diff
@@ -0,0 +1,44 @@
+/* eslint-disable @typescript-eslint/no-explicit-any */
+import autobind from 'autobind-decorator';
+import { Response } from 'express';
+
+import { C100Applicant } from '../../../../app/case/definition';
+import { AppRequest } from '../../../../app/controller/AppRequest';
+import { AnyObject, PostController } from '../../../../app/controller/PostController';
+import { Form, FormFields, FormFieldsFn } from '../../../../app/form/Form';
+import { getApplicantDetails, transformFormData, updateApplicantDetails } from '../util';
+
+import { getFormFields } from './content';
+
+@autobind
+export default class ContactPreferencePostController extends PostController<AnyObject> {
+  constructor(protected readonly fields: FormFields | FormFieldsFn) {
+    super(fields);
+  }
+
+  public async post(req: AppRequest<AnyObject>, res: Response): Promise<void> {
+    const applicantId = req.params.applicantId as C100Applicant['id'];
+    const form = new Form(getFormFields().fields as FormFields);
```

Comment:
@giorgi-hmcts Can you please refactor this to the fields from session ? this has been done for all dynamic fields screen in below PR.
https://github.com/hmcts/prl-citizen-frontend/pull/674